rust/connlib/tunnel/src/io.rs

```diff
@@ -153,14 +153,21 @@ impl Io {
 
         if let Some(timeout) = self.timeout.as_mut() {
             if timeout.poll_unpin(cx).is_ready() {
-                let deadline = timeout.deadline().into();
+                // Always emit `now` as the timeout value.
```

Comment:
IMO it might be a little clearer to keep `deadline` and just assert that `deadline > now` before continuing.

Having a deadline in the past seems... weird, but maybe it can happen if NTP updates a clock that has drifted.

Reply:
Or I guess this would handle clock changes gracefully, so no strong feelings.

Reply:
> Having a deadline in the past seems... weird, but maybe it can happen if NTP updates a clock that has drifted.

`Instant`s are guaranteed to be monotonic so there cannot be a clockdrift.

Reply:
> Or I guess this would handle clock changes gracefully, so no strong feelings.

That was my intention. If it happens for whatever reason, I don't want to crash `connlib`. Time should always go forward and the timer resolving means that _at least_ the deadline has been reached. Calling the state machine with the current time is IMO more correct because time has in fact already advanced to that (and not only to the deadline).